### PR TITLE
feat(api): implement find users endpoint

### PR DESCRIPTION
**Pull request by kduret**
## Description

implement find users endpoint

**Fixes** MON-12228

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

### CODE REVIEW

File: src/Core/Application/Configuration/User/UseCase/FindUsers/FindUsers.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    /**
+     * @param User[] $users
+     * @return FindUsersResponse
+     */
+    public function createResponse(array $users): FindUsersResponse
+    {
+        $response = new FindUsersResponse($users);
+
+        return $response;
```
Comment (adr-mo) [inline src/Core/Application/Configuration/User/UseCase/FindUsers/FindUsers.php:70]:
```suggestion
        return new FindUsersResponse($users);
```

File: src/Core/Application/Configuration/User/Repository/ReadUsersRepositoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
```
Comment (adr-mo) [inline src/Core/Application/Configuration/User/Repository/ReadUsersRepositoryInterface.php:4]:
```suggestion
 * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
```

Comment (adr-mo) [reply src/Core/Application/Configuration/User/Repository/ReadUsersRepositoryInterface.php:4]: Same for all headers

File: src/Core/Application/Configuration/User/UseCase/FindUsers/FindUsers.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
```
Comment (adr-mo) [inline src/Core/Application/Configuration/User/UseCase/FindUsers/FindUsers.php:4]:
```suggestion
 * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
```

File: src/Core/Application/Configuration/User/UseCase/FindUsers/FindUsersErrorResponse.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
```
Comment (adr-mo) [inline src/Core/Application/Configuration/User/UseCase/FindUsers/FindUsersErrorResponse.php:4]:
```suggestion
 * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
```

File: src/Core/Application/Configuration/User/UseCase/FindUsers/FindUsersPresenterInterface.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
```
Comment (adr-mo) [inline src/Core/Application/Configuration/User/UseCase/FindUsers/FindUsersPresenterInterface.php:4]:
```suggestion
 * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
```

File: src/Core/Infrastructure/Configuration/User/Repository/DbReadUsersRepository.php [hunk 6]
```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
```
Comment (adr-mo) [inline src/Core/Infrastructure/Configuration/User/Repository/DbReadUsersRepository.php:4]:
```suggestion
 * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
```

File: tests/php/Core/Infrastructure/Configuration/User/Repository/DbUserFactoryTest.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
```
Comment (adr-mo) [inline tests/php/Core/Infrastructure/Configuration/User/Repository/DbUserFactoryTest.php:4]:
```suggestion
 * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
```

**Review by adr-mo — APPROVED** [review]

File: src/Core/Application/Configuration/User/Repository/ReadUsersRepositoryInterface.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Application\Configuration\User\Repository;
+
+use Core\Domain\Configuration\User\Model\User;
+
+interface ReadUsersRepositoryInterface
```
Comment (jeremyjaouen) [inline src/Core/Application/Configuration/User/Repository/ReadUsersRepositoryInterface.php:28]:
```suggestion
interface ReadUserRepositoryInterface
```

Comment (jeremyjaouen) [reply src/Core/Application/Configuration/User/Repository/ReadUsersRepositoryInterface.php:28]: same modification for the filename

File: src/Core/Infrastructure/Configuration/User/Repository/DbReadUsersRepository.php [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+namespace Core\Infrastructure\Configuration\User\Repository;
+
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+use Centreon\Infrastructure\RequestParameters\SqlRequestParametersTranslator;
+use Core\Infrastructure\Configuration\User\Repository\DbUserFactory;
+use Core\Application\Configuration\User\Repository\ReadUsersRepositoryInterface;
+
+class DbReadUsersRepository extends AbstractRepositoryDRB implements ReadUsersRepositoryInterface
```
Comment (jeremyjaouen) [inline src/Core/Infrastructure/Configuration/User/Repository/DbReadUsersRepository.php:33]:
```suggestion
class DbReadUserRepository extends AbstractRepositoryDRB implements ReadUsersRepositoryInterface
```

same for filename

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**